spec/jobs/threat_metrix_js_verification_job_spec.rb

```diff
@@ -0,0 +1,192 @@
+require 'rails_helper'
+
+RSpec.describe ThreatMetrixJsVerificationJob, type: :job do
+  let(:proofing_device_profiling_collecting_enabled) { true }
+  let(:threatmetrix_org_id) { 'ABCD1234' }
+  let(:threatmetrix_session_id) { 'some-session-id' }
+
+  let(:threatmetrix_signing_key) do
+    OpenSSL::PKey::RSA.new 2048
+  end
+
+  let(:threatmetrix_signing_cert_expiry) { Time.now + 3600 }
+
+  let(:threatmetrix_signing_certificate) do
+    name = OpenSSL::X509::Name.parse('/CN=signing')
+
+    cert = OpenSSL::X509::Certificate.new
+    cert.version = 2
+    cert.serial = 0
+    cert.not_before = Time.now
+    cert.not_after = threatmetrix_signing_cert_expiry
+
+    cert.public_key = threatmetrix_signing_key.public_key
+    cert.subject = name
+    cert.issuer = name
+
+    cert.sign threatmetrix_signing_key, 'SHA1'
+
+    cert.to_pem
+  end
+
+  let(:js) do
+    <<~END
+      console.log('Javascrqipt!');
+    END
+  end
+
+  let(:signature) do
+    if threatmetrix_signing_certificate.present?
+      cert = OpenSSL::X509::Certificate.new threatmetrix_signing_certificate
+      sig = threatmetrix_signing_key.sign 'SHA256', js
+      sig.unpack1('H*')
```

Comment:
wait can we replace the entire `.hex.split` etc etc thing with `pack` or `unpack`?